cpp/src/arrow/python/arrow_to_python.cc

```diff
@@ -122,6 +123,12 @@ Status GetValue(PyObject* context, const Array& arr, int64_t index, int32_t type
     case Type::DOUBLE:
       *result = PyFloat_FromDouble(static_cast<const DoubleArray&>(arr).Value(index));
       return Status::OK();
+    case Type::DATE64: {
+      PyDateTime_IMPORT;
```

Comment:
Ideally this should be executed in the highest scope possible within this compilation unit, let me have a tinker